doc/melange.rst

```diff
@@ -4,49 +4,188 @@
+
+.. code:: bash
+
+    output
+    ├── melange__App.js
+    └── lib
+        ├── lib__.js
+        ├── lib__Helper.js
+        └── lib.js
+
+``<module_system>`` specifies the JavaScript import and export formats used. The
+values allowed are ``es6`` and ``commonjs``.
+
+  - ``es6`` will follow `JavaScript modules <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules>`_,
+    and will produce imports like ``import * from 'foo'`` and exports like
+    ``export let ....``
+  - ``commonjs`` will follow `CommonJS modules <https://nodejs.org/api/modules.html>`_,
+    and will produce imports like ``require('foo')`` and exports like ``module.exports = ....``
+
+``<optional-fields>`` are:
+
+- ``(entries <modules>)`` specifies what modules will be built with Melange. By
+  default, Dune will use all the ``.ml/.re`` files in the same directory as the
+  ``dune`` file. This includes ones present in the file system as well
+  as ones generated by user rules. You can restrict this list by using a
+  ``(entries <modules>)`` field. ``<modules>`` uses the :ref:`ordered-set-language`,
+  where elements are module names and don't need to start with an uppercase
+  letter. For instance, to exclude module ``Foo``, use ``(entries (:standard \
+  foo))``
+
+- ``(libraries <library-dependencies>)`` specifies the library's dependencies.
+  Melange libraries can only use the simple form, like
+  ``(libraries foo pkg.bar)``. Keep in mind the following limitations:
+  - The ``re_export`` and ``select`` forms are not
+  supported.
+
+- ``(package <package>)`` allows to define to which JavaScript package the
+  artifacts produced by the ``melange.emit`` stanza will belong to. This is
+  useful when the ``libraries`` field includes a public library, as it helps
+  making  sure the modules from the library and the entry modules that are part
+  of the ``melange.emit`` stanza belong to the same resulting JavaScript
+  package.
+
+- ``(preprocess <preprocess-spec>)`` specifies how to preprocess files when
+  needed. The default is ``no_preprocessing``, and other options are described
+  in the :ref:`preprocessing-spec` section.
+
+- ``(preprocessor_deps (<deps-conf list>))`` specifies extra preprocessor
+  dependencies preprocessor, i.e., if the preprocessor reads a generated file.
+  The specification of dependencies is described in the :ref:`deps-field`
+  section.
+
+- ``(flags <flags>)`` specifies compilation flags to be passed to ``melc``, the
+  command line tool for the Melange compiler that produces the ``.js`` artifacts.
+  ``<flags>`` is specified in the :ref:`ordered-set-language`. It also supports
+  ``(:include ...)`` forms. The default value for is taken from the environment,
+  as a result it's recommended to write ``(flags ...)`` fields as
+  follows: ``(flags (:standard <my options>))``
+
+Limitations
+===========
+
+- JavaScript "monorepos" are not supported, so the use case of having multiple
```

Comment:
why are they not supported? Dune should be able to build them, then it's up to the user to decide how to depend on them (at runtime). Am I missing something?

Reply:
Hm, you are right. As we have not tested this scenario yet, I wanted to mention something like "here be dragons". 

For now I removed the whole paragraph in https://github.com/ocaml/dune/pull/6384/commits/3d4ca7104e53ac6a79b19d10b269b0df349921f9, once we test this use case, maybe add some template, we can document it properly.